view.js: emit event payload in a consistently named property

go-oleg opened the pull request:
For 'emit:mark', 'emit:tick', and 'emit:points' events, always put the payload for (e.g. mark, time, points) into a property called 'payload'.

The change in https://github.com/juttle/juttle/pull/567 requires making changes in a few places that receive marks and pass them on to somewhere else (https://github.com/juttle/juttle-service/issues/71 and https://github.com/juttle/juttle-client-library/issues/80). Since these places just pass-through the core payload of the event and only transform the surrounding event metadata, using a generic 'payload' allows us to future-proof against any changes we might make to the 'payload' contents. These intermediate steps shouldn't need to be changed every time - only the actual end-subscribers to the events (generally the views).

We can use 'data' instead of 'payload' if you guys think thats a more appropriate name.

@demmer @dmajda @mstemm

dmajda commented:
+1

I like `data` more than `payload`, but can live with both.

dmajda commented:
Just noticed in the commit message:

> For 'emit:mark', 'emit:tick', and 'emit:points' events

I think you wanted s/emit/view/.

demmer commented:
FWIW I'd prefer `data` as well but am ok with either. 

-m


> On Mar 22, 2016, at 7:16 AM, David Majda <notifications@github.com> wrote:
> 
> +1
> 
> I like data more than payload, but can live with both.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>

go-oleg commented:
Changed to `data` (and changed the formal arg name in the consumers to `event` instead of `data` to avoid `data.data`).

And thx for catching the commit msg typo.
